src/firmware_new/boards/frankie_v1/udp_multicast.c

```diff
@@ -0,0 +1,117 @@
+#include "firmware_new/proto/control_fw.pb.h"
+#include "lwip/api.h"
+#include "lwip/inet.h"
+#include "lwip/ip_addr.h"
+#include "lwip/memp.h"
+#include "lwip/opt.h"
+#include "lwip/pbuf.h"
+#include "lwip/sys.h"
+#include "lwip/tcp.h"
+#include "lwip/udp.h"
+#include "main.h"
+#include "pb.h"
+#include "pb_decode.h"
+#include "pb_encode.h"
+
+// TODO the messages are global for now, will need to be moved
+// when this file is properly integrated. This file only serves
+// as a reference implementation of reliable multicast with IPv6
+robot_ack ack       = robot_ack_init_zero;
+control_msg control = control_msg_init_zero;
+
+// this buffer is used to hold serialized proto. Allocated
+// on the data segment to avoid mallocing on every packet received
+static uint8_t buffer[robot_ack_size];
+
+// the thread that handles multicast recv/send
+static void udp_multicast_thread(void *arg);
+
+struct multicast_config
+{
+    ip_addr_t multicast_address;
+    unsigned multicast_port;
+    unsigned send_port;
+} typedef multicast_config_t;
+
+/*
+ * Thread that creates a send and recv socket, joins the specified
+ * multicast group and start listening for packets. Increments a counter
+ * on every multicast packet and sends a unicast packet back to the sender.
+ *
+ */
+static void udp_multicast_thread(void *arg)
+{
+    multicast_config_t *config = (multicast_config_t *)arg;
+    struct netbuf *rx_buf, *tx_buf;
+    int msg_count = 0;
+
+    // TODO proper err handling, for now all errs are ignored
```

Comment:
Make a ticket for the TODO?

Reply:
im hesitating to make TODO's because im not even sure how these errors will be handled

Reply:
I think its still good to link to a ticket even if you don't know exactly how you'll do it yet. It makes the TODO more actionable and provides a place where more info can be given, and worst case helps point to the person to talk to if someone else stumbles across it.

Reply:
added

Reply:
No strong preference then